subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformWithDependenciesIntegrationTest.groovy

```diff
@@ -351,6 +352,7 @@ project(':common') {
         )
     }
 
+    @Ignore // TODO: understand what's going on in this test
```

Comment:
The error is:

```
	> Cannot choose between the following variants of test:test:1.2:
          - runtime
          - test
        All of them match the consumer attributes:
          - Variant 'runtime' capability test:test:1.2:
              - Unmatched attributes:
                  - Provides color 'blue' but the consumer didn't ask for it
                  - Provides org.gradle.status 'release' but the consumer didn't ask for it
          - Variant 'test' capability test:test:1.2:
              - Unmatched attributes:
                  - Provides color 'orange' but the consumer didn't ask for it
                  - Provides org.gradle.status 'release' but the consumer didn't ask for it
```

As I understand it:

- We have a dependency on project `a` itself
- There is a dependency from project `a` to `test:test:1.2`
- That (simulated) `test:test:1.2` project has 2 outgoing variants with `color = blue` and `color = orange`
- All outgoing variants of our project should include `color = purple`

- We define an transform named external to see `color = blue` as `color = purple` - this should let `a` see `test:test:1.2` as having `color = purple`
- We define a transform named local to see `color = purple` as `color = green` - this should let us see project `a` as having `color = green`

- Project `a` ends up with multiple potential matches for `test:test:1.2`, `runtime` (blue) and `test` (orange), since we didn't request a specific color which would distinguish them.

If I change the test to only allow simulate a single variant of `test:test:1.2` by commenting out `.variant('test', [color: 'orange'])`, it passes. 

So the core issue seems to be `a` is not requesting a color.

(If I change `.variant('test', [color: 'orange'])` to `.variant('test', [color: 'green'])` so that there is definitely a `green` variant of `test:test:1.2` available the test also fails:

```
      > Cannot choose between the following variants of test:test:1.2:
          - runtime
          - test
        All of them match the consumer attributes:
          - Variant 'runtime' capability test:test:1.2:
              - Unmatched attributes:
                  - Provides color 'blue' but the consumer didn't ask for it
                  - Provides org.gradle.status 'release' but the consumer didn't ask for it
          - Variant 'test' capability test:test:1.2:
              - Unmatched attributes:
                  - Provides color 'green' but the consumer didn't ask for it
                  - Provides org.gradle.status 'release' but the consumer didn't ask for it
```)

If I run the resolvable configurations report on `a`, I see:

```
> Task :a:resolvableConfigurations
--------------------------------------------------
Configuration extender (l)
--------------------------------------------------

Extended Configurations
    - implementation

--------------------------------------------------
Configuration implementation (l)
--------------------------------------------------

Attributes
    - color = blue
```

So I think the issue is that the new `extender` configuration (since we shouldn't be resolving `implementation` directly, we resolve this instead, which extends `implementation`) somehow isn't getting the `color = blue` attribute added to it.  This seems like an `allprojects` ordering issue in the setup.  

I think the fix/workaround should be to add a `color` attribute to the `extender` configuration when it is created.  I'm going to experiment more with this tomorrow.

Reply:
Fixed it by doing this.